common/ffs.h

```diff
@@ -740,15 +741,19 @@ typedef struct CPD_EXTENTION_HEADER_ {
 #define CPD_EXT_TYPE_IOM_METADATA            24
 #define CPD_EXT_TYPE_MGP_METADATA            25
 #define CPD_EXT_TYPE_TBT_METADATA            26
-#define CPD_LAST_KNOWN_EXT_TYPE              CPD_EXT_TYPE_TBT_METADATA
+#define CPD_EXT_TYPE_GMF_CERTIFICATE         30
+#define CPD_EXT_TYPE_GMF_BODY                31
+#define CPD_EXT_TYPE_KEY_MANIFEST_EXT        34
+#define CPD_EXT_TYPE_SIGNED_PACKAGE_INFO_EXT 35
+#define CPD_EXT_TYPE_SPS_PLATFORM_ID         50
 
 typedef struct CPD_EXT_SIGNED_PACKAGE_INFO_MODULE_ {
     UINT8  Name[12];
     UINT8  Type;
     UINT8  HashAlgorithm;
     UINT16 HashSize;
     UINT32 MetadataSize;
-    UINT8  MetadataHash[32];
+    UINT8  MetadataHash[1]; // Can be 32 or 48 bit
```

Comment:
Use `MetadataHash[];`. Even MSVC supports this today, and GCC can optimise out any access to this array with indices larger than 0 anyway.

Reply:
I'm really unsure it's supported in anything that can still target Windows XP, and I still want to.

Reply:
Yeah, rewrote them with sizeof offsets, looks good too.